src/main/edit_delete/preProcess.ts

```diff
@@ -1,11 +1,47 @@
-const preProcess = (wordList: Array<any>, duration: number): Array<any> => {
-  for (let i = 0; i < wordList.length - 1; i += 1) {
-    wordList[i].duration = wordList[i + 1].start_time - wordList[i].start_time;
-    wordList[i].deleted = false;
+import { Transcription } from 'sharedTypes';
+import JSONTranscription from './JSONTranscription';
+import TranscriptWord from './TranscriptWord';
+
+const preProcessTranscript = (
+  jsonTranscript: any,
+  duration: number
+): Transcription => {
+  const numberOfWords: number = jsonTranscript.words.length;
+
+  const processedTranscript: Transcription = new JSONTranscription(
+    [],
+    jsonTranscript.confidence
+  );
+
+  for (let i = 0; i < numberOfWords - 1; i += 1) {
+    // duration includes the white space between current and next word
+    const wordDuration =
+      jsonTranscript[i + 1].start_time - jsonTranscript.words[i].start_time;
```

Comment:
`.words` is missing here after `jsonTranscript` which is causing the function to crash 

should be `jsonTranscript.words[i + 1]`

Reply:
should be fixed now